nfs/nfs.go

```diff
@@ -0,0 +1,90 @@
+package nfs
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/pkg/errors"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+const (
+	nfsmountGlobalSection  = "NFSMount_Global_Options"
+	nfsmountDefaultVersion = "Defaultvers"
+
+	defaultNFSVersionMajor = 4
+	defaultNFSVersionMinor = 0
+)
+
+// GetSystemDefaultNFSVersion reads the system default NFS version. This config can be overridden by nfsmount.conf under
+// configPrefix. If configPrefix is empty, it will be /etc by default.
+func GetSystemDefaultNFSVersion(configPrefix string) (major, minor int, err error) {
+	if configPrefix == "" {
+		configPrefix = "/etc"
+	}
+
+	configMap, err := getSystemNFSMountConfigMap(configPrefix)
+	if os.IsNotExist(err) {
+		return defaultNFSVersionMajor, defaultNFSVersionMinor, nil
+	} else if err != nil {
+		return 0, 0, errors.Wrap(err, "failed to read default NFS version from nfsmount.conf")
+	}
+	if globalSection, ok := configMap[nfsmountGlobalSection]; ok {
+		if configured, ok := globalSection[nfsmountDefaultVersion]; ok {
+			majorStr, minorStr, _ := strings.Cut(configured, ".")
+			if major, err = strconv.Atoi(majorStr); err != nil {
+				return 0, 0, fmt.Errorf("invalid NFS Version '%s'", configured)
+			} else if minorStr == "" {
+				minor = 0
+			} else if minor, err = strconv.Atoi(minorStr); err != nil {
+				return 0, 0, fmt.Errorf("invalid NFS Version '%s'", configured)
+			}
+			return major, minor, nil
+		}
+	}
+	return defaultNFSVersionMajor, defaultNFSVersionMinor, nil
+}
+
+// getSystemNFSMountConfigMap reads the nfsmount.conf under configPrefix. The returned result is in
+// map[section]map[key]value structure, and the global options is result["NFSMount_Global_Options"]. Refer to man page
+// for more detail: man 5 nfsmount.conf
+func getSystemNFSMountConfigMap(configPrefix string) (map[string]map[string]string, error) {
+	configFilePath := filepath.Join(configPrefix, "nfsmount.conf")
+	configFile, err := os.Open(configFilePath)
+	if err != nil {
+		return nil, err
+	}
+	defer configFile.Close()
+
+	configMap := make(map[string]map[string]string)
+	scanner := bufio.NewScanner(configFile)
+	var section = ""
+	var sectionMap map[string]string
+
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		if line == "" || strings.HasPrefix(line, "#") {
+			continue
+		}
+		if strings.HasPrefix(line, "[") {
+			section = strings.TrimSpace(strings.Trim(line, "[]"))
+			continue
+		}
+		if sectionMap = configMap[section]; sectionMap == nil {
+			sectionMap = make(map[string]string)
+			configMap[section] = sectionMap
+		}
+		if key, val, isParsable := strings.Cut(line, "="); isParsable {
+			sectionMap[strings.TrimSpace(key)] = strings.TrimSpace(val)
+		} else {
+			return nil, fmt.Errorf("invalid key-value pair: '%s'", line)
```

Comment:
The function fails if a line does not contain an `=`. Would there be such a case?

Reply:
I don't think so. The defconfig files under `/boot` are generated from `make menuconfig` before build the kernel, and all the items must conform `CONFIG_${key}=${value}` format.

Ref: https://wiki.gentoo.org/wiki/Kernel/Gentoo_Kernel_Configuration_Guide

Reply:
Got it. Thank you!